service/s3/internal/customizations/handle_200_error.go

```diff
@@ -0,0 +1,126 @@
+package customizations
+
+import (
+	"bytes"
+	"context"
+	"encoding/xml"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"strings"
+
+	"github.com/awslabs/smithy-go"
+	"github.com/awslabs/smithy-go/middleware"
+	smithyhttp "github.com/awslabs/smithy-go/transport/http"
+	smithyxml "github.com/awslabs/smithy-go/xml"
+)
+
+// HandleResponseErrorWith200Status check for S3 200 error response.
+// If an s3 200 error is found, status code for the response is modified temporarily to
+// 5xx response status code.
+func HandleResponseErrorWith200Status(stack *middleware.Stack) {
+	stack.Deserialize.Insert(&processResponseFor200ErrorMiddleware{}, "OperationDeserializer", middleware.After)
+	stack.Deserialize.Insert(&reset200StatusCodeMiddleware{}, "OperationDeserializer", middleware.Before)
+}
+
+// middleware to process raw response and look for error response with 200 status code
+type processResponseFor200ErrorMiddleware struct{}
```

Comment:
You don't have to do this, but I wonder if this could be a single middleware registered to multiple stack steps? Theoretically such a middleware could maintain its own internal state so metadata passing isn't necessary.

Reply:
The same middleware helper would need to be registered multiple times which is not supported with the middleware stack steps. Using the metadata to communicate to the `before` middleware probably is a good way to upstream that information.